webapp/src/app/user.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable, NgZone } from '@angular/core';
+import { AngularFireAuth } from 'angularfire2/auth';
+import { AngularFireDatabase, FirebaseObjectObservable } from 'angularfire2/database';
+import { Observable } from 'rxjs/Observable';
+import { Profile } from './types';
+
+// TODO: the following is temporary -- check with deast@
+import * as firebase from 'firebase/app';
+
+// these need to be strings, because we can't use enum in template
+// see details: https://github.com/angular/angular/issues/2885
+export type AuthStatus = "Unknown"|"LoggedIn"|"LoggedOut";
+
+@Injectable()
+export class UserService {
```

Comment:
I think the name here is a bit confusing cuz it also deals with state--- also what is the difference between observable and FirebaseObjectObservable?